src/dbnode/persist/fs/files.go

```diff
@@ -74,25 +74,27 @@ type FileSetFile struct {
 	AbsoluteFilepaths []string
 
 	CachedSnapshotTime time.Time
+	CachedSnapshotID   []byte
 	filePathPrefix     string
 }
 
-// SnapshotTime returns the SnapshotTime for the given FileSetFile. Value is meaningless
-// if the the FileSetFile is a flush instead of a snapshot.
-func (f *FileSetFile) SnapshotTime() (time.Time, error) {
-	if !f.CachedSnapshotTime.IsZero() {
+// SnapshotTimeAndID returns the snapshot time and id for the given FileSetFile.
+// Value is meaningless if the the FileSetFile is a flush instead of a snapshot.
```

Comment:
Can we validate it's a snapshot fileset file with a name check or something to return an error for this case?